R/replace_na.R

```diff
@@ -1,19 +1,31 @@
 #' Replace missing values
 #'
-#' @param data A data frame.
-#' @param replace A named list given the value to replace `NA` with
-#'   for each column.
+#' @param data A data frame or vector.
+#' @param replace The value(s) to replace `NA` with. If `data`
+#'   is a vector, a vector with one or more replacement values.
+#'   If `data` is a data.frame, a named list providing such
+#'   replacement vectors for each of the named columns.
 #' @param ... Additional arguments for methods. Currently unused.
 #' @examples
 #' library(dplyr)
 #' df <- tibble(x = c(1, 2, NA), y = c("a", NA, "b"))
 #' df %>% replace_na(list(x = 0, y = "unknown"))
+#' df$x %>% replace_na(3)
 #' @export
-replace_na <- function(data, replace = list(), ...) {
+replace_na <- function(data, replace = NULL, ...) {
   UseMethod("replace_na")
 }
 #' @export
-replace_na.data.frame <- function(data, replace = list(), ...) {
+replace_na.default <- function(data, replace = NULL, ...) {
+  if (is_null(replace))
+    stop("If 'data' is a vector, 'replace' has to be provided.")
```

Comment:
Why?

I think instead it would be better to assert that `replace` has length 1

Reply:
Well, the two things that that can happen are the following. Pretty clear message:

```r
> data[1:5] <- NULL
Error in data[1:5] <- NULL: replacement has length zero
> data[1:5] <- 1:3
Warning message in data[1:5] <- 1:3:
“number of items to replace is not a multiple of replacement length”
```

If this now happens through our function here, we get:

```r
> replace_na(data)
Error in data[are_na(data)] <- replace: replacement has length zero
> replace_na(data, 1:3)
Warning message in data[are_na(data)] <- replace:
“number of items to replace is not a multiple of replacement length”
```

Of which I only think the last one is clear.

Reply:
But this might also happen: `data[1:6] <- 1:3`. We want to avoid accidentally recycling the `replace` value

Reply:
OK, done!